python/hail/docs/_ext/hail_doctest.py

```diff
@@ -193,6 +194,32 @@ def summarize(self, out, verbose=None):
         out(string_io.getvalue())
         return res
 
+    def get_code(self):
+        return self.code
+
+    def run(self, test, compileflags=None, out=None, clear_globs=True):
+        original_optionflags = self.optionflags
+        for examplenum, example in enumerate(test.examples):
+            # Merge in the example's options.
+            self.optionflags = original_optionflags
```

Comment:
OK, maybe I'm being stupid, but why can't you just use an local variable for optionsflags in the loop, one for each example?  That way, you don't have to mutate self.optionflags at all.